Assets/Scripts/SS3D/Systems/Tile/Connections/AdjacencyTypes/AdjacencyData.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace SS3D.Systems.Tile.Connections.AdjacencyTypes
+{
+    /// <summary>
+    /// Stores the type of adjacency as well as if it exists or not. Used by Adjacency Map.
+    /// </summary>
+    public readonly struct AdjacencyData
+    {
+        public AdjacencyData(TileObjectGenericType genericType, TileObjectSpecificType specificType, bool exists)
+        {
+            GenericType = genericType;
+            SpecificType = specificType;
+            Exists = exists;
+        }
+
+        public readonly TileObjectGenericType GenericType;
+        public readonly TileObjectSpecificType SpecificType;
+        public readonly bool Exists;
+
+        public override bool Equals(object other)
+        {
+            return other is AdjacencyData otherData && GenericType == otherData.GenericType && SpecificType == otherData.SpecificType &&
+                   Exists == otherData.Exists;
+        }
+
+        public override int GetHashCode()
```

Comment:
What's the point of this ?

Reply:
Gave a warning due to the functions Equals() being overridden, as they also expect GetHashCode() to be overridden.

Reply:
Okay so it was actually really not obvious for me that it was okay to just call base.GetHashCode(). Usually this is an issue, since when you override Equals, then you change the logic of what makes two object equals, and two equal object should have the same hash code, hence why the warning, since the logic in GetHashCode should reflect the logic in Equals.
BUT, in this case, that's fine, because it's a value type. 
As microsoft doc says : value types whose fields have equal values have equal hash codes.
So my question, is, can we just remove the two overrides ? 

You have two  options here : 
Option 1) : leave it as it is but add the following comments
- for Equals() : Is this really necessary ? It seems to do what the base equals implementation does.
- for GetHashCode : value types whose fields have equal values have equal hash codes, since Equals just check same type and if values are equals, it's okay to just call the base GetHashCode.

Option 2) Remove both override but check that everything still works as expected.

Reply:
Removed the overrides. I wasn't the original author of this code so I am not sure why it was introduced. Testing with it removed doesn't seem to make a difference